File: Tai_lieu/Code/22c2bignum.py
n=int(input())
a=list(map(int,input().split()))
cnt=0
for i in a:
	if i==0:
		cnt=1

lim=10**18
s=1
for i in a:
	s*=i
	if s>lim:
		s=-1
		break
if (cnt==1):
	print(0)
else:
	print(s)

File: Tai_lieu/Code/22c2bignum.cpp
#include <bits/stdc++.h>
using namespace std;
#define int long long
const int lim=1e18;
int a[100005];
signed main()
{
	ios_base::sync_with_stdio(false);cin.tie(0);cout.tie(0);
	int n; cin >> n;
	int cnt=0;
	for (int i=1;i<=n;i++) 
	{
		cin >> a[i];
		cnt+=(a[i]==0);
	}
	if (cnt) return cout << 0,0;
	int s=1;
	for (int i=1;i<=n;i++)
	{
		if (lim/a[i]>=s) s*=a[i];
		else
		{
			s=-1;
			break;
		}
	}
	cout << s;
}